apps/src/lib/node/ledger/protocol/mod.rs

```diff
@@ -216,6 +252,24 @@ where
     }
 }
 
+/// This is a helper function for getting the set of active validators for a
+/// given epoch.
+pub fn get_active_validators<D, H>(
```

Comment:
This function would be useful in the `queries.rs` module. In fact, it would probably simplify a lot of the existing functions in there. Just also make the epoch optional and if a `None` is passed in, use the current.

Reply:
:+1: I'll open a small PR to move this to `queries.rs` and some of the other voting power helper code so it can be shared between `Shell` and `apply_tx`